python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -49,6 +50,53 @@
 )
 from .partition_key_range import PartitionKeyRange
 
+POST_DST_TRANSITION_SUFFIX = "_POST_DST_TRANSITION"
+
+
+def dst_safe_strftime(dt: datetime, timezone_str: str, fmt: str) -> str:
```

Comment:
do we need to worry at all about this triggering incorrectly on daily partitions? e.g. if you have a partition that is set with timeranges at 2AM daily - one of them lands on an ambiguous date, but the partition key isn't actually ambigious since there is only one key on that day anyway

You might need to do some kind of 'repeats every hour' check like in the logic here: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_utils/schedules.py#L544-L549

and only trigger this new behavior for partitions that repeat every hour?